File: CSharp/Easy/Strings/LongestCommonPrefix.cs
/*
Example 1:

Input: ["flower","flow","flight"]
Output: "fl"
Example 2:

Input: ["dog","racecar","car"]
Output: ""
Explanation: There is no common prefix among the input strings.
*/

public class Solution {
    public string LongestCommonPrefix(string[] strs) {
        if(strs.Length==0)
        {
        	return "";
        }

        for(int i=0;i<strs[0].Length;i++)
        {
        	for(int j =0;j<strs.Length;j++)
        	{
        		if(i==strs[j].Length||strs[0][i]!=strs[j][i])
        		{
        			return strs[0].Substring(0,i);
        		}
        	}
        }
        return strs[0];
    }
}

File: Java/Easy/Strings/LongestCommonPrefix.java
class Solution {
    public String longestCommonPrefix(String[] strs) {
        if(strs.length==0)
        {
            return "";
        }
        String longestCommonPrefix = strs[0];
        int i=1;
        while(i<strs.length)
        {
        	if(strs[i].length()<longestCommonPrefix.length())
        	{
        		longestCommonPrefix = longestCommonPrefix.substring(0,strs[i].length());
        	}
            if(!strs[i].substring(0,longestCommonPrefix.length()).equals(longestCommonPrefix))
            {
               while(!strs[i].substring(0,longestCommonPrefix.length()).equals(longestCommonPrefix) && longestCommonPrefix.length()>0)
               {
                   longestCommonPrefix = longestCommonPrefix.substring(0,longestCommonPrefix.length()-1);
               }
                if(longestCommonPrefix.length()==0)
                {
                    return "";
                }
            }
            i++;
        }
        return longestCommonPrefix;
    }
}